client/app/scripts/superdesk-desks/aggregate.js

```diff
@@ -342,9 +341,19 @@
                     current: 'desks'
                 };
 
+                scope.$watch('step.current', function(step) {
```

Comment:
Can it happen that `step` is equal to `'searches'` from the very beginning?

If it can and if it is undesired to save searches even when nothing has, in fact, changed, then an additional check is needed here (`$watcher` is automatically run when registered for initialization reasons, even if there is no change).